addon/schema-manager.js

```diff
@@ -3,18 +3,58 @@ export class SchemaManager {
     this.schema = null;
   }
 
-  computeAttributeReference(key, value, modelname, schemaInterface) {
-    return this.schema.computeAttributeReference(key, value, modelname, schemaInterface);
+  /**
+   * Determines whether an attribute is a reference.
+   * If it is not, return `null` or `undefined`.
+   * Otherwise return an object with properties:
+   *  - `id` The id of the referenced model (either m3 or `DS.Model`)
+   *  - `type` The type of the referenced model (either m3 or `DS.Model`)
```

Comment:
`null` is also a valid type in which case `id` will be looked up in a global cache